packages/babel-helper-create-class-features-plugin/src/fields.js

```diff
@@ -497,7 +531,7 @@ export function buildFieldsInitNodes(
     const isField = prop.isProperty();
     const isMethod = !isField;
 
-    if (isStatic && isField) {
+    if (isStatic) {
       const replaced = replaceThisContext(prop, ref, superRef, state, loose);
```

Comment:
In static methods, we should only replace `super` and not `this`:

```js
class A {
  static get a() { return 1 }
}

class B {
  static get b() { return 2 }

  static #getA() { return super.a }
  static #getB() { return this.b }

  static extract() {
    return [this.#getA, this.#getB];
  }
}

const [getA, getB] = B.extract();
expect(getA.call({ a: 3 })).toBe(1);
expect(getB.call({ b: 4 })).toBe(4);
```

You can add a flag to `replaceThisContext` to toggle this behavior.

Reply:
Thanks, updated tests around this as well.